sdk/python/kfp/compiler/_container_builder.py

```diff
@@ -18,19 +18,54 @@
 import os
 import uuid
 
+SERVICEACCOUNT_NAMESPACE = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
+GCS_STAGING_BLOB_DEFAULT_PREFIX = 'kfp_container_build_staging'
+
 class ContainerBuilder(object):
   """
   ContainerBuilder helps build a container image
   """
-  def __init__(self, gcs_staging, namespace):
+  def __init__(self, gcs_staging=None, namespace=None):
     """
     Args:
-      gcs_staging (str): GCS blob that can store temporary build files
+      gcs_staging (str): GCS bucket/blob that can store temporary build files,
+          default is gs://PROJECT_ID/kfp_container_build_staging.
+      namespace (str): kubernetes namespace where the pod is launched,
+          default is the same namespace as the notebook service account in cluster
+              or 'kubeflow' if not in cluster
     """
-    if not gcs_staging.startswith('gs://'):
-      raise ValueError('Error: {} should be a GCS path.'.format(gcs_staging))
+    from pathlib import PurePath
     self._gcs_staging = gcs_staging
+    if gcs_staging is None:
+      gcs_bucket = self._get_project_id()
+      from ._gcs_helper import GCSHelper
+      GCSHelper.create_gcs_bucket_if_not_exist(gcs_bucket)
+      self._gcs_staging = 'gs://' + gcs_bucket + '/' + GCS_STAGING_BLOB_DEFAULT_PREFIX
+    else:
+      path = PurePath(self._gcs_staging).parts
+      if len(path) < 2 or not path[0].startswith('gs'):
+        raise ValueError('Error: {} should be a GCS path.'.format(gcs_staging))
+      from ._gcs_helper import GCSHelper
+      GCSHelper.create_gcs_bucket_if_not_exist(path[1])
```

Comment:
nit: you can reuse the create bucket code in both user provided and default gcs path.

Reply:
Done.